kitsune/sumo/static/sumo/js/switching-devices-wizard-manager.js

```diff
@@ -18,15 +26,105 @@ export default class SwitchingDevicesWizardManager {
     utm_campaign: "migration",
     utm_medium: "mozilla-websites",
 
+    // These are the other parameters that are sent through the FxA sign-in
+    // flow.
     entrypoint: "fx-new-device-sync",
     entrypoint_experiment: null,
     entrypoint_variation: null,
     flow_id: null,
     flow_begin_time: null,
     context: null,
+
+    // The rest of these are internal state variables used by this class
+    // to determine which step to show the user in the <form-wizard>.
+    fxaRoot: null,
+    fxaSignedIn: false,
+    sumoEmail: "",
+    syncEnabled: false,
+    confirmedSyncChoices: false,
   };
 
-  constructor(formWizard) {
+  #state = Object.assign({}, this.#defaultState);
+
+  /**
+   * @typedef {object} SwitchingDevicesWizardStep
+   * @property {string} name
+   *   The name of the step to pass to <form-wizard>.setStep if this
+   *   SwitchingDevicesWizardStep is the one we're entering.
+   * @property {function} exitConditionsMet
+   *   A function that accepts the state object from
+   *   SwitchingDevicesWizardManager. Returns true the
+   *   SwitchingDevicesWizardStep can be skipped over because the
+   *   conditions for completing it are met.
+   * @property {function} enter
+   *   A function that accepts the state object from
+   *   SwitchingDevicesWizardManager. Returns another object that
+   *   is passed to the <form-wizard>.setStep method as the payload
+   *   for configuring the displayed step.
+   */
+
+  // An array of SwitchingDevicesWizardSteps that are evaluated during each
+  // state change to determine which step we should put the user in with the
+  // <form-wizard>. This is intentionally public so that the steps can be
+  // unit tested individually.
+  steps = Object.freeze([
+    {
+      name: "sign-into-fxa",
```

Comment:
Just a note (maybe to self) - we'll need to make sure the `name` attributes in `hook_device_migration_wizard.html` match what we've got here in order to get things working. Doesn't necessarily have to be done here though.

Reply:
Thanks! Yeah, I put these in as placeholders for now. If we decide there are better names, updating these and the tests is fairly straight-forward.